tools/openiris_setup.py

```diff
@@ -0,0 +1,816 @@
+            elif "error" in pause_response and pause_response["error"] == "Command timeout":
+                # Even if we timeout, the command likely worked (as seen in logs)
+                print("✅ Device pause command sent (startup logs may have obscured response)")
+            else:
+                print(f"⚠️  Pause status uncertain: {pause_response}")
+            
+            return True
+        except Exception as e:
+            print(f"❌ Failed to connect to {self.port}: {e}")
+            return False
+    
+    def disconnect(self):
+        """Disconnect from the device"""
+        if self.connection and self.connection.is_open:
+            # Optionally unpause the device before disconnecting
+            print("Resuming device startup...")
+            self.send_command("pause", {"pause": False})
+            
+            self.connection.close()
+            print(f"🔌 Disconnected from {self.port}")
+    
+    def send_command(self, command: str, params: Dict = None, timeout: int = None) -> Dict:
+        """Send a command to the device and wait for response"""
+        if not self.connection or not self.connection.is_open:
+            return {"error": "Not connected"}
+        
+        cmd_obj = {"commands": [{"command": command}]}
+        if params:
+            cmd_obj["commands"][0]["data"] = params
+        
+        cmd_json = json.dumps(cmd_obj) + '\n'
+        
+        try:
+            # Clear any pending data
+            self.connection.reset_input_buffer()
+            
+            # Send command
+            print(f"📤 Sending: {cmd_json.strip()}")
+            self.connection.write(cmd_json.encode())
+            
+            # For scan_networks command, handle special case
+            if command == "scan_networks":
+                # Use provided timeout or default to 30 seconds for scan
+                scan_timeout = timeout if timeout is not None else 30
+                return self._handle_scan_response(scan_timeout)
+            
+            # Wait for response (skip heartbeats and logs)
+            start_time = time.time()
+            response_buffer = ""
+            
+            # Use provided timeout or default to 15 seconds
+            cmd_timeout = timeout if timeout is not None else 15
+            while time.time() - start_time < cmd_timeout:
+                try:
+                    if self.connection.in_waiting:
+                        data = self.connection.read(self.connection.in_waiting).decode('utf-8', errors='ignore')
+                        response_buffer += data
+                        
+                        # Show raw data for debugging
+                        if self.debug and data.strip():
+                            print(f"📡 Raw: {repr(data)}")
+                            print(f"📝 Buffer: {repr(response_buffer[-200:])}")
+                        
+                        # Clean buffer and look for JSON
+                        import re
```

Comment:
We can keep imports like that at the very top, there are special cases where keeping them on top of a function makes sense, but never inside of the body